packages/ui/src/AddressView/index.tsx

```diff
@@ -5,14 +5,27 @@ import { bech32mAddress } from '@penumbra-zone/bech32m/penumbra';
 import { CopyToClipboardButton } from '../CopyToClipboardButton';
 import { AddressIcon } from './AddressIcon';
 import { Text } from '../Text';
-import { useDensity } from '../utils/density';
+import { TextVariant } from '../Text/types';
+import { useDensity, Density as DensityType } from '../utils/density';
 import { Density } from '../Density';
 
+const textVariantByDensity = (density: DensityType): TextVariant => {
+  if (density === 'slim') {
+    return 'detail';
+  }
+  if (density === 'compact') {
+    return 'small';
+  }
+  return 'strong';
+};
+
 export interface AddressViewProps {
   addressView: AddressView | undefined;
   copyable?: boolean;
   hideIcon?: boolean;
   truncate?: boolean;
+  /** If true, takes the `altBech32m` field from the address and renders it as-is */
+  external?: boolean;
```

Comment:
added support for AddressView component to render addresses from other blockchains like `osmo1awefasdf...`